source/common/http/path_utility.cc

```diff
@@ -54,35 +95,37 @@ bool PathUtil::canonicalPath(RequestHeaderMap& headers) {
   if (!query_suffix.empty()) {
     normalized_path.insert(normalized_path.end(), query_suffix.begin(), query_suffix.end());
   }
-  headers.setPath(normalized_path);
-  return true;
+  return normalized_path;
 }
 
-void PathUtil::mergeSlashes(RequestHeaderMap& headers) {
-  ASSERT(headers.Path());
-  const auto original_path = headers.getPathValue();
-  // Only operate on path component in URL.
-  const absl::string_view::size_type query_start = original_path.find('?');
-  const absl::string_view path = original_path.substr(0, query_start);
-  const absl::string_view query = absl::ClippedSubstr(original_path, query_start);
-  if (path.find("//") == absl::string_view::npos) {
-    return;
+PathTransformer::PathTransformer(envoy::type::http::v3::PathTransformation path_transformation) {
+  const auto& operations = path_transformation.operations();
+  std::vector<uint64_t> operation_hashes;
```

Comment:
Depending on the length, it might be better to use `absl::flat_hash_set`.

Reply:
It might be overkill to use a hash set here since we expect a very small number of transformation.